core/utils/poly-look/src/visualisations/charts/bar-charts/horizontalBarChart/horizontalBarChart.js

```diff
@@ -0,0 +1,329 @@
+      .transition()
+      .delay(1000)
+      .duration(500)
+      .attr("fill", this._groups ? this._barValueColor : "black");
+  }
+
+  _addEnteringGroupLabels(groupLabels) {
+    groupLabels
+      .append("text")
+      .attr(
+        "y",
+        (d) =>
+          this._yScales
+            .find((scaleContainer) => scaleContainer.id === d.id)
+            .scale.range()[0]
+      )
+      .attr("class", "group-label")
+      .attr("x", 0)
+      .text((d) => d.translation)
+      .attr("fill", "transparent")
+      .style("font-size", "15px")
+      .transition()
+      .delay(1000)
+      .duration(500)
+      .attr("fill", "black");
+  }
+
+  _addEnteringBarValues(barGroups) {
+    barGroups
+      .append("text")
+      .attr("y", (d) => {
+        for (let scaleContainer of this._yScales) {
+          if (!scaleContainer.id)
+            return (
+              scaleContainer.scale(d.title) +
+              this._barWidth +
+              barTextBottomMargin / 2
+            );
+          if (scaleContainer.id === d.group)
+            return (
+              scaleContainer.scale(d.title) +
+              scaleContainer.scale.bandwidth() -
+              barTextBottomMargin
+            );
+        }
+      })
+      .attr("class", "bar-value")
+      .attr("text-anchor", "end")
+      .attr("x", (d) => this._xScale(d.value) - barValueMargin)
+      .text((d) => d.value)
+      .attr("fill", "transparent")
+      .style("font-size", "10px")
+      .transition()
+      .delay(1000)
+      .duration(500)
+      .attr("fill", this._barValueColor);
+  }
+
+  _updateExistingBarValues(barValues) {
+    barValues
+      .attr("fill", "transparent")
+      .attr("x", 0)
+      .text((d) => d.value)
+      .raise()
+      .transition()
+      .delay(1500)
+      .duration(500)
+      .attr("fill", this._barValueColor);
+  }
+
+  _alignBarText() {
+    const xScale = this._xScale;
+    const groups = this._groups;
+    const barColor = this._barColor;
+    this.chart.selectAll(".bar-group").each(function () {
+      const node = d3.select(this);
+      const value = node.select(".bar-value");
+      const label = node.select(".bar-label");
+      const labelWidth = groups ? label.node().getBBox().width : 0;
+      if (
+        value.node().getBBox().width + labelWidth + barValueMargin >
+        xScale(+value.node().textContent)
+      ) {
+        if (groups)
+          label
+            .attr("text-anchor", "start")
+            .attr("x", (d) => barValueMargin + xScale(d.value))
+            .attr("fill", "transparent")
+            .transition()
+            .delay(1000)
+            .duration(500)
+            .attr("fill", barColor);
+        value
+          .attr("text-anchor", "start")
+          .attr(
+            "x",
+            (d) =>
+              barValueMargin + xScale(d.value) + barValueMargin + labelWidth
+          )
+          .attr("fill", "transparent")
+          .transition()
+          .delay(1000)
```

Comment:
Again, attributes repeated from above. Possibly have a function that adds this common code. Except for the `x` attribute, the rest is the same, right?